source/Calamari/Kubernetes/Conventions/GatherAndApplyRawYamlConvention.cs

```diff
@@ -0,0 +1,172 @@
+            {
+                var files = Directory.GetFiles(deployment.CurrentDirectory, glob);
+                var directory = Path.Combine(deployment.CurrentDirectory, "grouped", idx.ToString());
+                Directory.CreateDirectory(directory);
+                foreach (var file in files)
+                {
+                    fileSystem.CopyFile(file, Path.Combine(directory, Path.GetFileName(file)));
+                }
+
+                directories.Add(directory);
+            }
+
+            variables.Set(SpecialVariables.GroupedYamlDirectories, string.Join(";", directories));
+            return directories;
+        }
+
+        private IEnumerable<Resource> ApplyBatchAndReturnResources(int index, string glob, Kubectl kubectl, string directory)
+        {
+            log.Info($"Applying Batch #{index} for YAML matching '{glob}'");
+            var result = kubectl.ExecuteCommandAndReturnOutput("apply", "-f", directory, "-o", "json");
+
+            foreach (var message in result.Output.Messages)
+            {
+                switch (message.Level)
+                {
+                    case Level.Info:
+                        //No need to log as it's the output json from above.
+                        break;
+                    case Level.Error:
+                        log.Error(message.Text);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            if (result.Result.ExitCode != 0)
+            {
+                throw new Exception(result.Result.Errors);
+            }
+
+            // If it did not error, the output should be valid json.
+            var outputJson = result.Output.InfoLogs.Join(Environment.NewLine);
+            try
+            {
+                var token = JToken.Parse(outputJson);
+                var lastResources = token.Type == JTokenType.Array
+                    ? token.ToObject<List<Resource>>()
+                    : new List<Resource> { token.ToObject<Resource>() };
+
+                foreach (var resource in lastResources)
+                {
+                    log.Info($"'{resource.Kind}/{resource.Metadata.Name}' created.");
+                }
+
+                return lastResources;
+            }
+            catch
+            {
+                log.Error($"\"kubectl apply -o json\" returned invalid JSON for Batch #{index}:");
+                log.Error("---------------------------");
+                log.Error(outputJson);
+                log.Error("---------------------------");
+                log.Error("This can happen with older versions of kubectl. Please update to a recent version of kubectl.");
+                log.Error("See https://github.com/kubernetes/kubernetes/issues/58834 for more details.");
+                log.Error("Custom resources will not be saved as output variables.");
```

Comment:
This error message is based on the one found in the Deployment scripts that were originally used.